app/iot/service.py

```diff
@@ -45,5 +46,13 @@ def run_program(self, program: list[Message]) -> None:
             self.send_msg(msg)
```

Comment:
The 'send_msg' method is asynchronous, but it is being called synchronously here. This will likely cause issues since 'send_msg' should be awaited. Consider using 'await self.send_msg(msg)' within an async function or refactor this method to handle asynchronous calls properly.